packages/ui/src/components/DatePicker/SDatePickerPanelCalendars.vue

```diff
@@ -0,0 +1,146 @@
+<script setup lang="ts">
+import MonthTable from './SDatePickerTableMonths.vue'
+import YearTable from './SDatePickerTableYears.vue'
+import DatePanel from './SDatePickerPanelDate.vue'
+
+import { StateStore, ShowState, ModelValueType } from './types'
+import { DatePickerApi, useDatePickerApi } from './api'
+
+interface Props {
+  currentView: string
+  showState: ShowState
+  stateStore: StateStore
+  modelValue: ModelValueType
+}
+
+const props = defineProps<Props>()
+
+const state: DatePickerApi = useDatePickerApi()
+const { type, time } = state
+
+const emit = defineEmits(['updateShowedState', 'changeView', 'pick'])
+
+const hoveredDate = ref<Date>(new Date())
+
+const isRange = computed(() => {
+  return type === 'range'
+})
```

Comment:
Are you sure that destructuring `state` is fine for reactivity?

I mean, it's fine if `state` is not reactive for sure. But then you don't need to make `isRange` a `computed`, because `type` is a static value.

I think here is a mistake somewhere.